integrations/iron-server-core/src/main/scala/sttp/iron/server/iron/TapirServerIron.scala

```diff
@@ -0,0 +1,33 @@
+package sttp.iron.server.iron
+
+import sttp.tapir.DecodeResult
+import sttp.tapir.DecodeResult.Error
+import sttp.tapir.DecodeResult.Error.JsonDecodeException
+import sttp.tapir.server.interceptor.DecodeFailureContext
+import sttp.tapir.server.interceptor.decodefailure.DefaultDecodeFailureHandler.FailureMessages
+import sttp.tapir.server.interceptor.decodefailure.{DecodeFailureInterceptor, DefaultDecodeFailureHandler}
+
+trait TapirServerIron {
+
+  case class IronException(errorMessage: String) extends Exception(errorMessage)
```

Comment:
Which exception do you mean? Iron throws `java.lang.IllegalArgumentException` when we just try to `.refine` he value instead of `.refineEither`, but I don't think that's the one?

Reply:
I think I might be missing a usage example ;)

Reply:
now that I understand more I think that maybe there's too many moving parts to provide a useful integration. We've got the iron <-> json library integration, and then the integration <-> tapir server integration. Maybe it would be simples to create an example for a single, specific combination, and briefly describe it in the docs, instead of providing a module?

Reply:
As sad as it is to throw away the written code - I quite agree. In the iron <-> tapir server integration we provide the exception, and failure handler handling the exception, but we can't provide the middle part - which is tapir and server interpreter. It kind of feels like having 2 loose ends without the connection between them. 

The failure handler and custom exception are useful though and worth documenting with example code.

Reply:
Throwing away code is part of the process, nothing to be sad about :)